src/OpenTelemetry/Metrics/MetricProvider.cs

```diff
@@ -0,0 +1,147 @@
+// <copyright file="MetricProvider.cs" company="OpenTelemetry Authors">
+// Copyright The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// </copyright>
+
+using System;
+using System.Collections.Concurrent;
+using System.Collections.Generic;
+using System.Diagnostics.Metrics;
+using System.Threading;
+using System.Threading.Tasks;
+
+#nullable enable
+
+namespace OpenTelemetry.Metrics
+{
+    public class MetricProvider : IDisposable
```

Comment:
This is different from my understanding, I guess:
1. The name will be `MeterProvider`.
2. It will derive from [`OpenTelemetry.BaseProvider`](https://github.com/open-telemetry/opentelemetry-dotnet/blob/98055871511d0b56df75cb7443a1ed9a49918014/src/OpenTelemetry.Api/BaseProvider.cs#L24).
3. We might want to move something to the API? (e.g. considering if someone wants to implement their own SDK) @cijothomas might have better idea here.

Reply:
We can follow similar approach as tracing. Will iterate over several PRs. :)